extensions/amp-geo/0.1/amp-geo.js

```diff
@@ -207,6 +215,115 @@ export class AmpGeo extends AMP.BaseElement {
         'GEONOTPATCHED: amp-geo served unpatched, ISO country not set'
       );
     }
+
+    return this.mode_ !== mode.GEO_API
+      ? Promise.resolve()
+      : this.fetchCountry_().then(country => {
+          if (country) {
+            this.country_ = country;
+          } else {
+            // if API request fails, leave the country at the default 'unknown'
+            this.error_ = true;
+            dev().error(
+              TAG,
+              'GEONOTPATCHED: amp-geo served unpatched and API response not valid, ISO country not set'
+            );
+          }
+        });
+  }
+
+  /**
+   * Ensure API URL definition is usable and cast its type
+   * @param {*} url
+   * @return {?string}
+   */
+  validateApiUrl(url) {
+    if (typeof url !== 'string') {
+      user().error(TAG, 'geoApiUrl must be a string URL');
+      return null;
+    }
+
+    if (!isSecureUrlDeprecated(url)) {
+      user().error(TAG, 'geoApiUrl must be secure (https)');
+      return null;
+    }
+
+    if (isProxyOrigin(url)) {
```

Comment:
ProxyOrigin doesn't work the same reason any random url doesn't work. I don't think we need a special check for this. Let me know if you think differently.

Reply:
Admittedly, there's not much reason for this check. It can be removed. I will wait for a final decision on `isSecureUrlDeprecated`/`assertHttpsUrl` before pushing a new change.

Reply:
I don't think it's needed.